libraries/shared/watcher/storage_watcher.go

```diff
@@ -108,7 +108,7 @@ func (watcher StorageWatcher) transformDiffs() error {
 			return getDiffErr
 		}
 
-		minID = int(diffID)
+		minID = int(diffID - 1)
 	}
```

Comment:
Just pushed this change up for now... wondering if a cleaner approach would be to change the `GetNewDiffs` query to look at diffs where id `>=` instead of just `>`, but think I need to look at that again with fresh eyes tomorrow.

Reply:
Makes sense to prefer `>=` to `>` 👍 

Not sure I follow why we need to set `minID` to `diffID - 1` though?

Reply:
Cool, i'll make update to `>=`. 
I needed to change the `minID` to `diffID - 1` because when passing in a block to start looking at diffs, we would use the id from the first diff for that block. Then if we pass that id to `GetNewDiffs` it would look up diffs with an id `>` than the given id, so that diff would never be processed. So by making the `minID` one less than that diff's id, it would ensure that it would get processed. Does that make sense?

All that being said, changing GetNewDiffs to be inclusive for the minID seems a lot easier.

Reply:
just writing out the 2 options again to help myself understand the two approaches:
1. if we keep `GetNewDiffs` as is with `id > minId` then we will need to pass in an id less than the diffId that we’re getting from `GetFirstDiffIDForBlockHeight` otherwise that id won’t ever be included in `GetNewDiffs`, so that is why I was passing in `firstDiffId -1`

1. on the flip side, if we change `GetNewDiffs` to do `id >= minId` then whenever we pass in an updated minID which is the last id from the set of diffs returned, then we’d end up returning that last diff from `GetNewDiffs` again. So it seems like we’d need to pass in `lastDiffId +1` as the new diff id